diff --git a/subprojects/gst-plugins-good/gst/matroska/matroska-read-common.c b/subprojects/gst-plugins-good/gst/matroska/matroska-read-common.c
index e1e35dc058..7996a88a65 100644
--- a/subprojects/gst-plugins-good/gst/matroska/matroska-read-common.c
+++ b/subprojects/gst-plugins-good/gst/matroska/matroska-read-common.c
@@ -2699,7 +2702,7 @@ gst_matroska_read_common_parse_metadata_id_tag (GstMatroskaReadCommon * common,
         GstMatroskaTrackContext *stream = g_ptr_array_index (common->src, j);
 
         if (stream->uid == tgt) {
-          gst_tag_list_insert (stream->tags, taglist, GST_TAG_MERGE_REPLACE);
+          gst_tag_list_insert (stream->tags, taglist, GST_TAG_MERGE_APPEND);
           stream->tags_changed = TRUE;
           found = TRUE;
         }
@@ -2711,7 +2714,7 @@ gst_matroska_read_common_parse_metadata_id_tag (GstMatroskaReadCommon * common,
         GstTagList *cached_taglist =
             g_hash_table_lookup (common->cached_track_taglists, track_uid);
         if (cached_taglist)
-          gst_tag_list_insert (cached_taglist, taglist, GST_TAG_MERGE_REPLACE);
+          gst_tag_list_insert (cached_taglist, taglist, GST_TAG_MERGE_APPEND);
         else {
           gst_tag_list_ref (taglist);
           g_hash_table_insert (common->cached_track_taglists, track_uid,
