diff --git a/subprojects/gstreamer/libs/gst/base/gstbaseparse.c b/subprojects/gstreamer/libs/gst/base/gstbaseparse.c
index 422e489c32..9a6eaa8640 100644
--- a/subprojects/gstreamer/libs/gst/base/gstbaseparse.c
+++ b/subprojects/gstreamer/libs/gst/base/gstbaseparse.c
@@ -2290,7 +2290,7 @@ gst_base_parse_handle_buffer (GstBaseParse * parse, GstBuffer * buffer,
       GST_BUFFER_PTS (outbuf) = pts;
       GST_BUFFER_DTS (outbuf) = dts;
       GST_BUFFER_OFFSET (outbuf) = GST_BUFFER_OFFSET_NONE;
-      GST_BUFFER_DURATION (outbuf) = GST_CLOCK_TIME_NONE;
+      // GST_BUFFER_DURATION (outbuf) = GST_CLOCK_TIME_NONE;
       GST_BUFFER_OFFSET_END (outbuf) = GST_BUFFER_OFFSET_NONE;
       GST_BUFFER_FLAGS (outbuf) = 0;
       parse->priv->buffers_head =
@@ -3316,7 +3316,7 @@ gst_base_parse_chain (GstPad * pad, GstObject * parent, GstBuffer * buffer)
       tmpbuf = gst_buffer_make_writable (tmpbuf);
       GST_BUFFER_PTS (tmpbuf) = parse->priv->next_pts;
       GST_BUFFER_DTS (tmpbuf) = parse->priv->next_dts;
-      GST_BUFFER_DURATION (tmpbuf) = GST_CLOCK_TIME_NONE;
+      // GST_BUFFER_DURATION (tmpbuf) = GST_CLOCK_TIME_NONE;
     }
 
     /* keep the adapter mapped, so keep track of what has to be flushed */
