diff --git a/subprojects/gst-plugins-bad/ext/srtp/meson.build b/subprojects/gst-plugins-bad/ext/srtp/meson.build
index 1d543a224e..3ea6cf6026 100644
--- a/subprojects/gst-plugins-bad/ext/srtp/meson.build
+++ b/subprojects/gst-plugins-bad/ext/srtp/meson.build
@@ -23,6 +23,8 @@ plugin_sources += {
   'srtp': pathsep.join(doc_sources)
 }
 
+gst_plugins_install_dir = join_paths(get_option('libdir'), 'gstreamer-1.0')
+
 srtp_cargs = []
 srtp_option = get_option('srtp').enable_if(get_option('webrtc').enabled(), error_message: 'webrtc option is enabled')
 if srtp_option.disabled()
@@ -30,7 +32,7 @@ if srtp_option.disabled()
   subdir_done()
 endif
 
-srtp_dep = dependency('libsrtp2', version : '>= 2.1.0', required : srtp_option, allow_fallback: true)
+srtp_dep = dependency('libSRTP', modules: ['libSRTP::srtp2'], version : '>= 2.1.0', required : false)
 
 if srtp_dep.found()
   gstsrtp_enums = gnome.mkenums_simple('gstsrtp-enumtypes',
@@ -45,7 +47,7 @@ if srtp_dep.found()
     include_directories : [configinc],
     dependencies : [gstrtp_dep, gstvideo_dep, srtp_dep],
     install : true,
-    install_dir : plugins_install_dir,
+    install_dir : gst_plugins_install_dir,
   )
   plugins += [gstsrtp]
 endif
